openstack/resource_openstack_compute_instance_v2.go

```
@@ -599,6 +648,57 @@ func resourceComputeInstanceV2Update(d *schema.ResourceData, meta interface{}) e
 		}
 	}
 
+	if d.Get("state").(string) != "" {
+		if d.HasChange("state") {
+			vmState := d.Get("state").(string)
+			if strings.ToLower(vmState) == "shutoff" {
+				err = startstop.Stop(computeClient, d.Id()).ExtractErr()
+				if err != nil {
+					log.Printf("[WARN] Error stopping OpenStack instance: %s", err)
+				} else {
+					stopStateConf := &resource.StateChangeConf{
+						//Pending:    []string{"ACTIVE"},
+						Target:     []string{"SHUTOFF"},
+						Refresh:    ServerV2StateRefreshFunc(computeClient, d.Id()),
+						Timeout:    3 * time.Minute,
+						Delay:      10 * time.Second,
+						MinTimeout: 3 * time.Second,
+					}
+
+					log.Printf("[DEBUG] Waiting for instance (%s) to stop", d.Id())
+					_, err = stopStateConf.WaitForState()
+					if err != nil {
+						log.Printf("[WARN] Error waiting for instance (%s) to stop: %s, proceeding to delete", d.Id(), err)
+					}
+				}
+
+			}
+			if strings.ToLower(vmState) == "active" {
+				err = startstop.Start(computeClient, d.Id()).ExtractErr()
+				if err != nil {
+					log.Printf("[WARN] Error starting OpenStack instance: %s", err)
```

We should return an error rather than log one.